src/receive_buffer.cpp

```diff
@@ -317,15 +316,15 @@ int crypto_receive_buffer::advance_pos(int bytes)
 	return sub_transferred;
 }
 
-buffer::const_interval crypto_receive_buffer::get() const
+span<char const> crypto_receive_buffer::get() const
 {
-	buffer::const_interval recv_buffer = m_connection_buffer.get();
+	span<char const> recv_buffer = m_connection_buffer.get();
 	if (m_recv_pos < m_connection_buffer.pos())
-		recv_buffer.end = recv_buffer.begin + m_recv_pos;
+		recv_buffer = recv_buffer.subspan(0, m_recv_pos);
```

Comment:
Unless I'm reading wrong, I don't think it's the same. `first` seems to discard the last `n` elements, could you confirm?